port/zos390/omrvmem.c

```diff
@@ -1276,17 +1276,21 @@ get_protectionBits(uintptr_t mode)
 void
 omrvmem_default_large_page_size_ex(struct OMRPortLibrary *portLibrary, uintptr_t mode, uintptr_t *pageSize, uintptr_t *pageFlags)
 {
-	uint32_t index = 0;
+	int32_t indexPageable = -1;
+	int32_t index = -1;
+	BOOLEAN pageablePreferableRequested = ((NULL != pageFlags) && (IS_VMEM_PAGE_FLAG_PAGEABLE_PREFERABLE(*pageFlags)));
 
 	if ((NULL == pageSize) &&
 		(NULL == pageFlags)
 	) {
 		return;
 	}
+
 	if (NULL != pageSize) {
 		*pageSize = 0;
 	}
-	if (NULL != pageFlags) {
+
+	if ((NULL != pageFlags) && (FALSE == pageablePreferableRequested)) {
```

Comment:
you don't need to check ` && (FALSE == pageablePreferableRequested)` here any more. The only reason use to be to not overwrite Pageable Preferable flag. Now it is cached in `pageablePreferableRequested`. This would solve side effect asymmetry in line 1329 where `* pageFlags` might be uninitialized at all

Reply:
Right. That's not required anymore, will be removed.